packages/gatsby/src/redux/machines/page-component.ts

```diff
@@ -121,17 +165,19 @@ module.exports = Machine(
           } else {
             return ctx.pages
           }
-        },
+        }
       }),
       deletePage: assign({
         pages: (ctx, event) => {
-          ctx.pages.delete(event.page.path)
+          if (event.page) {
+            ctx.pages.delete(event.page.path)
+          }
```

Comment:
`DELETE_PAGE` should always have `page` so this check is really unnecessary other than just to satisfy typechecker

I would say let's use non-null assertion operator here to not add runtime cost
```suggestion
          ctx.pages.delete(event.page!.path)
```

Reply:
I added your suggestion but I had to add an eslint-ignore `no-non-null-assertion` above the line as well. Even with that, it makes more sense this way.

Reply:
I think the `eslint-ignore` is related to the other problem you have that for some reason eslint doesn't use our config. 

We use this operator in other files ( for example https://github.com/gatsbyjs/gatsby/blob/4fb4b3addfa9134de0cd0085084e8c00a3d2e98a/packages/gatsby/src/redux/nodes.ts#L573 ) and we don't need `ignore` directive there (or I'm missing one)

Reply:
I removed it, thank you for highlighting it. I'll keep trying to investigate.

I think my vscode could have issues with the monorepo format, using the base config instead of the package specific one. But not sure.

Reply:
I just looked up - we actually do have warnings for those:
```
/home/circleci/project/packages/gatsby/src/utils/tracer/index.ts
  26:14  warning  Forbidden non-null assertion  @typescript-eslint/no-non-null-assertion
```

but not errors - I think it's fine to have this as a warning. Ideally at some point we could figure out tighter typings